packages/wrangler/src/proxy.ts

```diff
@@ -398,9 +404,15 @@ function createStreamHandler(
  */
 export async function waitForPortToBeAvailable(
   port: number,
-  options: { retryPeriod: number; timeout: number }
+  options: { retryPeriod: number; timeout: number; abortSignal: AbortSignal }
 ): Promise<void> {
   return new Promise((resolve, reject) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (options.abortSignal as any).addEventListener("abort", () => {
+      const error = new Error("");
```

Comment:
This will log "Failed to start server:" which will be confusing to anyone who reads it. Could we add a code or something this `error`, and read it on line 307 and not log anything if it's an aborted attempt? Like @petebacondarwin described in https://github.com/cloudflare/wrangler2/pull/732#pullrequestreview-927401192

Reply:
```suggestion
      const abortError = new Error("waitForPortToBeAvailable() aborted");
      (error as {code: string}).code = "ABORT_ERR";
```